simulation/simtypes/action.go

```diff
@@ -105,3 +104,32 @@ func (m msgBasedAction) Execute(sim *SimCtx, ctx sdk.Context) (
 	}
 	return sim.deliverTx(tx, msg, m.name)
 }
+
+func totalFrequency(actions []Action) int {
+	totalFrequency := 0
+	for _, action := range actions {
+		totalFrequency += mapFrequencyFromString(action.Frequency())
+	}
+
+	return totalFrequency
+}
+
+type selectActionFn func(r *rand.Rand) Action
+
+func GetSelectActionFn(actions []Action) selectActionFn {
+	totalOpFrequency := totalFrequency(actions)
+
+	return func(r *rand.Rand) Action {
+		x := r.Intn(totalOpFrequency)
+		// TODO: Change to an accum list approach
```

Comment:
What is that?

Reply:
This was a TODO Dev wrote previously and I moved it over, so I am not 100 percent sure